openproblems/tasks/label_projection/methods/seurat.py

```diff
@@ -0,0 +1,43 @@
+from ....tools.conversion import r_function
+from ....tools.decorators import method
+from ....tools.utils import check_r_version
+from typing import Optional
+
+import functools
+
+_seurat = r_function("seurat.R", args="sce, n_pcs, k_score=NULL, k_filter=NULL")
+
+_seurat_method = functools.partial(
+    method,
+    paper_name="Integrated analysis of multimodal single-cell data",
+    paper_url="https://doi.org/10.1016/j.cell.2021.04.048",
+    paper_year=2021,
+    code_url="https://github.com/satijalab/seurat",
+    image="openproblems-r-extras",
+)
+
+
+@_seurat_method(
+    method_name="Seurat reference mapping (SCTransform)",
+)
+def seurat(
+    adata,
+    n_pcs: Optional[int] = None,
+    k_score: Optional[int] = None,
+    k_filter: Optional[int] = None,
+    test: bool = False,
+):
+    kwargs = {}
+    if test:
+        kwargs["n_pcs"] = n_pcs or 5
+        kwargs["k_score"] = k_score or 5
+        kwargs["k_filter"] = k_filter or 20
+    else:  # pragma: nocover
+        kwargs["n_pcs"] = n_pcs or 50
+        if k_score is not None:
+            kwargs["k_score"] = k_score
+        if k_filter is not None:
+            kwargs["k_filter"] = k_filter
```

Comment:
I'm not sure the `is not None` works here... what happens if the argument isn't passed to the `.R` file but is used in the file? This might be a question for @scottgigante-immunai as it relates to the R wrapper function.

Reply:
@LuckyMD defaults are set to `NULL` on [this line](https://github.com/openproblems-bio/openproblems/pull/533/files/f5ac6eec7cc75b510c263cb70d10c9742ab9de18#diff-9c47aacc0e8aee164341ffdbed234feb74f204b3bd7a0e0c689f8d27aaefe0e2R8), so that shouldn't be a problem.

Reply:
Ahh, okay. I missed that. Thanks!